ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -92,6 +92,13 @@
   private volatile boolean responseComplete = false;
   // marked as true if force close is required because close() was called.
   private volatile boolean forceClose = false;
+  // the ResponseStatus that will be sent (or has been sent) as a part of the response.
+  private volatile ResponseStatus responseStatus = ResponseStatus.Ok;
+  // the response metadata that was actually sent.
+  private volatile HttpResponse finalResponseMetadata = null;
```

Comment:
Why use `volatile` here? Is it only set by one thread and read by 1 or multiple other threads?